File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
       //using bfs we have as
        if(!root) return 0;
        queue<TreeNode*> q;
        q.push(root);
        int depth = 0;
        while(!q.empty()){
            ++depth;
            int sz = q.size();
            for(int i=0;i<sz;i++){
            TreeNode* node = q.front(); 
            q.pop();
            if(node->left!=nullptr){
                q.push(node->left);
            }
            if(node->right!=nullptr){
                q.push(node->right);
            }
        }
        }
        return depth;
    }
};

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

interface TreeNode{
    val:number;
    left:TreeNode | null;
    right:TreeNode | null;
}

function maxDepth(root: TreeNode | null): number {
   if(!root){
       return 0;
   }
    const q:TreeNode[] = [];
    q.push(root);
    let depth = 0;
    while(q.length>0){
        let sz = q.length;
        depth+=1;
        for(let i=0;i<sz;i++){
            const node = q.shift() as TreeNode;
            if(node.left){
                q.push(node.left);
            }
             if(node.right){
                q.push(node.right);
            }
        }
    }
    return depth;
};







